src/DynamoRevit/Models/RevitDynamoModel.cs

```diff
@@ -360,12 +419,101 @@ private void OnApplicationViewActivated(object sender, ViewActivatedEventArgs e)
 
         public override void OnEvaluationCompleted(object sender, EvaluationCompletedEventArgs e)
         {
+            Debug.WriteLine(ElementIDLifecycleManager<int>.GetInstance());
+
+            if (historicalElementData.ContainsKey(CurrentWorkspace.Guid))
+            {
+                ReconcileHistoricalElements();
+            }
+
             // finally close the transaction!
             TransactionManager.Instance.ForceCloseTransaction();
 
             base.OnEvaluationCompleted(sender, e);
         }
 
+        private void ReconcileHistoricalElements()
+        {
+            // Read the current run's trace data into a dictionary which matches
+            // the layout of our historical element data. 
+
+            var traceData = EngineController.LiveRunnerCore.RuntimeData.GetTraceDataForNodes(
+                CurrentWorkspace.Nodes.Select(n => n.GUID),
+                EngineController.LiveRunnerCore.DSExecutable);
+
+            var currentRunData = new Dictionary<Guid, List<int>>();
+            foreach (var kvp in traceData)
+            {
+                var idList = new List<int>();
+                currentRunData.Add(kvp.Key, idList);
+
+                foreach (var serializables in kvp.Value.Select(DeserializeCallsiteData)) {
+                    idList.AddRange(serializables.Select(ser => ((SerializableId)ser).IntID));
+                }
+            }
+
+            // Compare the historical element data and the current run
+            // data to see whether there are elements that had been
+            // created by nodes previously, which were not created in 
+            // this run. Store the elements ids of all the elements that
+            // can't be found in the orphans collection.
+
+            var workspaceHistory = historicalElementData[CurrentWorkspace.Guid];
+            var orphanedIds = new List<int>();
+
+            foreach (var kvp in workspaceHistory)
+            {
+                var nodeGuid = kvp.Key;
+
+                // If the current run doesn't have a key for 
+                // this guid, then all of these elements are 
+                // orphaned.
+
+                if (!currentRunData.ContainsKey(nodeGuid))
+                {
+                    orphanedIds.AddRange(kvp.Value);
+                    continue;
+                }
+
+                var currentRunNodeGuids = currentRunData[nodeGuid];
+
+                // If the current run didn't create an element
+                // in the current run, then add an orphan.
+
+                orphanedIds.AddRange(kvp.Value.Where(id => !currentRunNodeGuids.Contains(id)));
```

Comment:
@ikeough 

I just checked the logic related to trace data and verified that the current run trace data will not include those that are left over. For example, if we change from 0..3 to 0..1, the current run trace data will only contain 2 instead of 4.

Based on this, your current logic is correct.